pkg/manager/internal.go

```diff
@@ -389,17 +370,18 @@ func (cm *controllerManager) serveMetrics(stop <-chan struct{}) {
 		Handler: mux,
 	}
 	// Run the server
-	go func() {
+	cm.startRunnable(RunnableFunc(func(stop <-chan struct{}) error {
```

Comment:
plumb this as context instead of stop channel?  Or do we want to save that for a followup?

Reply:
That breaks the api so must go into a 0.7.0, whereas this PR does not I believe, hence I would prefer to do it in a different PR

Reply:
+1 plumbing context as a follow-on and keeping API as-is.